src/Features/CSharp/Portable/CodeFixes/Async/CSharpAddAwaitCodeFixProvider.cs

```diff
@@ -76,8 +72,19 @@ private bool IsCorrectReturnType(ExpressionSyntax expression, SemanticModel sema
 
         private static ExpressionSyntax ConvertToAwaitExpression(ExpressionSyntax expression)
         {
-            return SyntaxFactory.AwaitExpression(expression)
-                                .WithAdditionalAnnotations(Formatter.Annotation);
+            AwaitExpressionSyntax result;
+
+            if (expression.HasLeadingTrivia)
```

Comment:
@jmarolf Two questions:

1. Shouldn't the first line of your code be `..(expression.WithoutTrivia())`?
1. It seems that our code does not handle adding parenthesis(the only case I can think of that needs propagate the trailing trivia) when needed

For example:
```diff
- return true ? Task.Delay(0) : Task.Delay(1) /* what? */;
+ return await (true ? Task.Delay(0) : Task.Delay(1)) /* what? */;
```

Problems with the above case:

1. VS2015 RTM shows CS4016 but does not provide a fix
1. The comment looks wrongly propagated for me